components/proxy/plugins/configcat/configcat.go

```diff
@@ -94,9 +123,14 @@ func (c *ConfigCat) Provision(ctx caddy.Context) error {
 	c.configCache = make(map[string]*configCache)
 
 	c.sdkKey = os.Getenv("CONFIGCAT_SDK_KEY")
+	c.fromConfigMap = os.Getenv("CONFIGCAT_FROM_CONFIGMAP") == "true"
```

Comment:
Given a change to ENV variables on a container will always cause the pod to restart, you could cache this at plugin startup time to avoid looking it up during each request. Unless you want the application to dynamically respond to updated env variables from the env, which in itself generally doesn't work as one would expect.